File: Internet Banking/src/java/com/internet_banking/User_registration.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.internet_banking;

import java.io.IOException;
import java.io.PrintWriter;
import java.util.Base64;
import java.util.Date;
import javax.servlet.ServletException;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import javax.servlet.http.HttpSession;
/**
 *
 * @author Sumanpc
 */
public class User_registration extends HttpServlet {

    /**
     * Processes requests for both HTTP <code>GET</code> and <code>POST</code>
     * methods.
     *
     * @param request servlet request
     * @param response servlet response
     * @throws ServletException if a servlet-specific error occurs
     * @throws IOException if an I/O error occurs
     */
    protected void processRequest(HttpServletRequest request, HttpServletResponse response)
            throws ServletException, IOException {
        response.setContentType("text/html;charset=UTF-8");
        try (PrintWriter out = response.getWriter()) {
            /* TODO output your page here. You may use following sample code. */
            String actype,fname,lname,ftname,dob,gender,occupation,annual_income,brname,brcode,country,state,
                    district,locality,zip_code,email,Phone_no;
                actype=request.getParameter("actype");
                fname= request.getParameter("fname");
                lname= request.getParameter("lname");
                ftname= request.getParameter("ftname");
                dob= request.getParameter("dob");
                gender= request.getParameter("gender");
                occupation= request.getParameter("occupation");
                annual_income= request.getParameter("annual_income");
                brname= request.getParameter("brname");
                brcode= request.getParameter("brcode");
                country= request.getParameter("country");
                state= request.getParameter("state");
                district= request.getParameter("district");
                locality= request.getParameter("locality");
                zip_code= request.getParameter("zip_code");
                email= request.getParameter("email");
                Phone_no= request.getParameter("Phone_no");
                Date date= new Date();
                long time = date.getTime();
               //String cif_no = String.valueOf(Math.random()).replaceAll("^\\d*\\.","");
                //Password = Password.substring(Password.indexOf("."));
                
                UserRegistrationGetSet ugs = new UserRegistrationGetSet();
                  ugs.setActype(actype);
                  ugs.setFname(fname);
                  ugs.setLname(lname);
                  ugs.setFtname(ftname);
                  ugs.setDob(dob);
                  ugs.setGender(gender);
                  ugs.setOccupation(occupation);
                  ugs.setAnnual_income(annual_income);
                  ugs.setBrname(brname);
                  ugs.setBrcode(brcode);
                  ugs.setCountry(country);
                  ugs.setState(state);
                  ugs.setDistrict(district);
                  ugs.setLocality(locality);
                  ugs.setZip_code(zip_code);
                  ugs.setEmail(email);
                  ugs.setPhone_no(Phone_no);
                  ugs.setTime(time);
                  ugs.setType("user");
                  HttpSession session =request.getSession();
                  session.setAttribute("mobile",Phone_no);
                  //time=Long.parseLong("1580024318694");
                  Databass d1 = new Databass();
                  int x=d1.insertuser_Details(ugs);
                  if(x==1)
                  {
                      String timeVal=String.valueOf(time);
                      String msg=Base64.getEncoder().encodeToString(timeVal.getBytes());
                        //String msg= org.apache.tomcat.util.codec.binary.Base64.encodeBase64String(timeVal.getBytes());
                       response.sendRedirect("Bankers/NewUserDocUpload.jsp?msg="+msg);
                  }
                  else
                  {
//                      String timeVal=String.valueOf(time);
//                      String msg=Base64.getEncoder().encodeToString(timeVal.getBytes());
                      response.sendRedirect("Bankers/AcForNew_user.jsp?Error=");
                  }
        }
        catch(Exception ex)
        {
            
        }
    }

    // <editor-fold defaultstate="collapsed" desc="HttpServlet methods. Click on the + sign on the left to edit the code.">
    /**
     * Handles the HTTP <code>GET</code> method.
     *
     * @param request servlet request
     * @param response servlet response
     * @throws ServletException if a servlet-specific error occurs
     * @throws IOException if an I/O error occurs
     */
    @Override
    protected void doGet(HttpServletRequest request, HttpServletResponse response)
            throws ServletException, IOException {
        processRequest(request, response);
    }

    /**
     * Handles the HTTP <code>POST</code> method.
     *
     * @param request servlet request
     * @param response servlet response
     * @throws ServletException if a servlet-specific error occurs
     * @throws IOException if an I/O error occurs
     */
    @Override
    protected void doPost(HttpServletRequest request, HttpServletResponse response)
            throws ServletException, IOException {
        processRequest(request, response);
    }

    /**
     * Returns a short description of the servlet.
     *
     * @return a String containing servlet description
     */
    @Override
    public String getServletInfo() {
        return "Short description";
    }// </editor-fold>

}

File: Internet Banking/web/Java_Script/User_js1/User_Registration.js
/* 
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
/* global doucment */
function user_validation1()
{
    var actype = document.getElementById("actype");
    var fname=document.getElementById('fname');
    var lname=document.getElementById('lname');
    var ftname=document.getElementById('ftname');
    var dob= document.getElementById('dob');
    var gender=document.getElementById('gender');
    var occupation=document.getElementById('occupation');
    var annual_income=document.getElementById('annual_income');
    var brname=document.getElementById('brname');
    var brcode=document.getElementById('brcode');
    var country=document.getElementById('country');
    var state=document.getElementById('state');
    var district=document.getElementById('district');
    var locality=document.getElementById('locality');
    var zip_code=document.getElementById('zip_code');
    var email=document.getElementById('email');
    var Phone_no=document.getElementById('Phone_no');
   
    var f=0;
    if(actype.value === "")
    {  
        actype.style.borderColor="red";
        document.getElementById('cus1').innerHTML="Please Enter The Account Type";
      
        f=1;
    }
    else {
        actype.style.borderColor="green";
        document.getElementById('cus1').innerHTML="";
        
    }
    if(fname.value === "")
    {  
            fname.style.borderColor="red";
            document.getElementById('cus2').innerHTML="Please Enter The First Name";
            f=1;
        
    }
    else 
    {
         if(fname.value.length < 3)
        {
            fname.style.borderColor="red";
            document.getElementById('cus2').innerHTML="Please Enter The Three Charter";
            f=1; 
        }
        else
        {
            fname.style.borderColor="green";
            document.getElementById('cus2').innerHTML="";
        }
        
    }
    if(lname.value === "")
    {  
        lname.style.borderColor="red";
        document.getElementById('cus3').innerHTML="Please Enter The Last Name";
        f=1;
    }
    else
    {
        if(lname.value.length<3)
        {
            lname.style.borderColor="red";
            document.getElementById('cus3').innerHTML="Please Enter The The Three Charter";
            f=1;
        }
        else
        {
            lname.style.borderColor="green";
            document.getElementById('cus3').innerHTML="";
        }
        
    }
    if(ftname.value === "")
    {  
        ftname.style.borderColor="red";
        document.getElementById('cus4').innerHTML="Please Enter The Father's Name";
        f=1;
    }
    else 
    {
        if(ftname.value.length<3)
        {
            ftname.style.borderColor="red";
            document.getElementById('cus4').innerHTML="Please Enter The Three Charter";
            f=1;
        }
        else
        {
            ftname.style.borderColor="green";
            document.getElementById('cus4').innerHTML="";
        }
        
    }
    if(dob.value === "")
    {  
        dob.style.borderColor="red";
        document.getElementById('cus5').innerHTML="Please Enter The Date of Birth";
      
        f=1;
    }
    else {
        dob.style.borderColor="green";
        document.getElementById('cus5').innerHTML="";
        
    }
    if(occupation.value === "")
    {  
        occupation.style.borderColor="red";
        document.getElementById('cus7').innerHTML="Please Enter The Applicant's Occupation";
      
        f=1;
    }
    else {
        occupation.style.borderColor="green";
        document.getElementById('cus7').innerHTML="";
        
    }
    if(annual_income.value === "")
    {  
        annual_income.style.borderColor="red";
        document.getElementById('cus8').innerHTML="Please Enter The Annual Income";
      
        f=1;
    }
    else {
        annual_income.style.borderColor="green";
        document.getElementById('cus8').innerHTML="";
        
    }
     if(brname.value === "")
    {  
        brname.style.borderColor="red";
        document.getElementById('cus9').innerHTML="Please Enter The Branch Name";
      
        f=1;
    }
    else {
        brname.style.borderColor="green";
        document.getElementById('cus9').innerHTML="";
        
    }
      if(brcode.value === "")
    {  
        brcode.style.borderColor="red";
        document.getElementById('cus10').innerHTML="Please Enter The Branch Code";
      
        f=1;
    }
    else {
        brcode.style.borderColor="green";
        document.getElementById('cus10').innerHTML="";
        
    }
     if(country.value === "")
    {  
        country.style.borderColor="red";
        document.getElementById('cus11').innerHTML="Please Enter The Country";
      
        f=1;
    }
    else {
        country.style.borderColor="green";
        document.getElementById('cus11').innerHTML="";
        
    }
     if(state.value === "")
    {  
        state.style.borderColor="red";
        document.getElementById('cus12').innerHTML="Please Enter The State";
      
        f=1;
    }
    else {
        state.style.borderColor="green";
        document.getElementById('cus12').innerHTML="";
        
    }
     if(district.value === "")
    {  
        district.style.borderColor="red";
        document.getElementById('cus13').innerHTML="Please Enter The District";
      
        f=1;
    }
    else {
        district.style.borderColor="green";
        document.getElementById('cus13').innerHTML="";
        
    }
     if(locality.value === "")
    {  
        locality.style.borderColor="red";
        document.getElementById('cus14').innerHTML="Please Enter The Locality";
      
        f=1;
    }
    else {
        locality.style.borderColor="green";
        document.getElementById('cus14').innerHTML="";
        
    }
     if(zip_code.value === "")
    {  
        zip_code.style.borderColor="red";
        document.getElementById('cus15').innerHTML="Please Enter The Zip Code";
      
        f=1;
    }
    else {
        zip_code.style.borderColor="green";
        document.getElementById('cus15').innerHTML="";
        
    }
     if(email.value === "")
    {  
        email.style.borderColor="red";
        document.getElementById('cus16').innerHTML="Please Enter The Email";
      
        f=1;
    }
    else {
        email.style.borderColor="green";
        document.getElementById('cus16').innerHTML="";
        
    }
     if(Phone_no.value === "")
    {  
        Phone_no.style.borderColor="red";
        document.getElementById('cus17').innerHTML="Please Enter The Phone Number";
      
        f=1;
    }
    else {
            if(Phone_no.value.length!=10)
            {
                Phone_no.style.borderColor="red";
                document.getElementById('cus17').innerHTML="The Mobile Number Must Be of 10 Digit";
                f=1;
            }
            else
            {
                Phone_no.style.borderColor="green";
                document.getElementById('cus17').innerHTML="";
            }
        
    }
    
    if(f===1)
    {
        return false;
    }
    else
    {
        return true;
    }
}

